### ISSUE TITLE
Randomly fadeOut glitch on Component hide

### ISSUE THREAD

**Issue by marcocavanna** [adammcarth/react-native-segmented-picker#15]
I open a new Issue to divide topics.

### Description
When the component hide using internal function and fadeOut animation occurs a random glitch showing the component for just half a second after fadeOut is completed.

#### Possibile Enhance
I don't know if problem is here, but as the `fadeOut` is a Promise, personally I'd prefer waiting for both end before setting visible to false.

So, instead writing:

```js
this.modalContainerRef.current.fadeOut(ANIMATION_TIME);
await this.pickerContainerRef.current.fadeOut(ANIMATION_TIME);
this.setState({ visible: false }, () => {
    this.cache.purge();
});
```

You probably could be more consistent and avoid any error writing
```js
await Promise.all([
  /** Hide Modal Container */
  new Promise((resolve) => {
    if (this.modalContainerRef.current?.fadeOut) {
      return this.modalContainerRef.current.fadeOut(ANIMATION_TIME).then(resolve);
    }
    return resolve();
  }),

  /** Hide Picker Container */
  new Promise((resolve) => {
    if (this.pickerContainerRef.current?.fadeOut) {
      return this.pickerContainerRef.current.fadeOut(ANIMATION_TIME).then(resolve);
    }
    return resolve();
  }),
]);

this.setState({ visible: false }, () => {
  this.cache.purge();
});
```

**Comment by adammcarth** [comment]
Hi @marcocavanna. I've been playing around with a few different options today - do you want to see if #16 solves the issues on your end? If it doesn't, I might need a little more info from you - because I can't seem to replicate it anymore.

Note that I originally did something similar to you with the `Promise.all()` technique, but I realised that I didn't actually need to fade both elements out anyway - only the parent container! Also, note that those manual `fadeOut` methods are only used by Android as of the latest version (`v1.1.0`). iOS relies on the default React Native `<Modal />` fade out functionality because I found smoother results using that instead.

**Comment by adammcarth** [comment]
This should be fixed in `v1.1.1`. Please re-open if not.
